### PR TITLE
Refresh README

### PR DESCRIPTION
**Pull request by aaron-lane**
## Background

To improve the user experience of this module, we are redesigning the README to focus on ease of getting started.

Motivation and discussion regarding these changes can be found in the [PRD](https://docs.google.com/document/d/1ljNbXGr0g40m7kzjwcNV47AONTGo1Ajza09YuhfuJg4/edit).

Relates #38 

## How Has This Been Tested

These documentation changes have not been tested.

## This PR makes me feel

This is too long. I decline to read it.

### CODE REVIEW

**Comment by aaron-lane** [comment]
@rogeruiz I tried to strike a balance between helping new users and reducing maintenance effort for us. My understanding is that the Registry renders inputs and outputs directly from the configuration so we don't need to own the extra logic required to generate markdown tables across each repository. To compromise for users who prefer to view the inputs and outputs on GitHub, I suggest that we decide on a consistent organization for the contents of `variables.tf` and `outputs.tf` which mimics the Registry layout.

More broadly, I tried to avoid including any content where similar content is provided by the Registry. My thinking is that less content which must be changed with each release will lead to easier  maintenance at scale as we want to keep the README structure consistent across the different modules.

**Comment by glasner** [comment]
I think this a great step forward. Hard for me to understand how the variables will look without seeing in the registry. I would say don't wait on me.. any changes I would suggest can be layered on later

File: README.md [hunk 1]
```diff
@@ -1,54 +1,136 @@
-# Terraform Enterprise: Clustering for Google
+# Terraform Enterprise Clustered on Google Cloud Platform
 
-![Terraform Logo](https://github.com/hashicorp/terraform-google-terraform-enterprise/blob/master/assets/TerraformLogo.png?raw=true)
+The **terraform-enterprise** Terraform module is the official way for
+HashiCorp customers to provision [Terraform Enterprise Clustered][tfe]
+on Google Cloud Platform (GCP).
 
-## Description
+![architecture diagram][architecture-diagram]
```
Comment (erindatkinson) [inline README.md:7]: This isn't a blocker for the review but this is a VERY high level diagram, and we may want to update these both in the repos as well as in the docs https://github.com/hashicorp/terraform-google-terraform-enterprise/pull/55/files#diff-04c6e90faac2675aa89e2176d2eec7d8R12 to be more precise.

File: README.md [hunk 2]
```diff
@@ -1,54 +1,136 @@
...
+used to provision this module:
+
+- Cloud SQL Admin (roles/cloudsql.admin)
+- Compute Admin (roles/compute.admin)
+- DNS Administrator (roles/dns.admin)
+- Service Account Admin (roles/iam.serviceAccountAdmin)
+- Service Account Key Admin (roles/iam.serviceAccountKeyAdmin)
+- Service Account User (roles/iam.serviceAccountUser)
+- Service Networking Admin (roles/servicenetworking.networksAdmin)
+- Storage Admin (roles/storage.admin)
```
Comment (erindatkinson) [inline README.md:43]: 💯, A+, Stellar 10/10 would recommend

File: README.md [hunk 3]
```diff
@@ -1,54 +1,136 @@
...
+[github-issues]: https://github.com/hashicorp/terraform-google-terraform-enterprise/issues
+[google-provider]: https://registry.terraform.io/providers/hashicorp/google
+[hashicorp-support]: https://support.hashicorp.com/
+[quickstart]: https://github.com/hashicorp/terraform-google-terraform-enterprise-quickstart
+[tf-community-forum]: https://discuss.hashicorp.com/c/terraform-core
+[tf-install]: https://learn.hashicorp.com/terraform/getting-started/install
+[tf-module-version]: https://www.terraform.io/docs/configuration/modules.html#module-versions
+[tf-registry]: https://registry.terraform.io/modules/hashicorp/terraform-enterprise/google
+[tfe-clustered-architecture]: https://www.terraform.io/docs/enterprise/before-installing/cluster-architecture.html
+[tfe]: https://www.terraform.io/docs/enterprise/index.html
```
Comment (erindatkinson) [inline README.md:136]: also stellar idea!

**Pull request merged**